pkgs/development/libraries/science/math/dealii/default.nix

```diff
@@ -0,0 +1,183 @@
+{ lib
+, stdenv
+, fetchzip
+, symlinkJoin
+, withDebug ? false
+, commonCxxFlags ? ""
+, debugCxxFlags ? ""
```

Comment:
Why do we need this? This can be easily done with overlays, can't it?

Reply:
This was for convenience. Adding CMake flags with spaces in their values isn't obvious ([my question at discourse](https://discourse.nixos.org/t/cmakeflags-and-spaces-in-option-values/20170)), so I wanted to add that to the derivation.

Reply:
but is that something that a regular user needs all the time or something rarely used?

Reply:
This library usually powers quite CPU-intensive software, so I may reasonably expect users to override flags with something like `-march=znver3`. A generic build wouldn't e.g. enable AVX, which can be used by the library and gives a noticeable performance boost.

Reply:
We should turn that on by default if ``stdenv.hostPlatform.avxSupport`` is true.

Also I think using overlays would be nicer and make the code less complex. Then we can also save that option entirely.

Reply:
> We should turn that on by default if `stdenv.hostPlatform.avxSupport` is true.

Isn't this option always `false`? On all machines I have access to with Nix it evaluates to `false` (as well as all SSE options), but all of them have sufficiently modern CPU.

Reply:
see https://nixos.wiki/wiki/Build_flags

Reply:
Thanks a lot, I'll look into that. Surprisingly, though, `system-features` cannot be overriden via `--option` on a NixOS system, but can be on a non-NixOS one. And do I understand correctly that these flags can only be set to **both** a package and its dependencies?

Reply:
I am not super familiar with that. Best to ask on matrix or discourse.

Reply:
After some thinking, it's best to leave all custom build flag stuff out of the derivation and document overriding elsewhere, like on the wiki.